pkg/addon/addon.go

```diff
@@ -1438,3 +1445,51 @@ func fetchVelaCoreImageTag(ctx context.Context, k8sClient client.Client) (string
 	}
 	return tag, nil
 }
+
+// PackageAddon package vela addon directory into a helm chart compatible archive and return its absolute path
+func PackageAddon(addonDictPath string) (string, error) {
+	meta := &Meta{}
+	metaData, err := ioutil.ReadFile(filepath.Clean(filepath.Join(addonDictPath, MetadataFileName)))
+	if err != nil {
+		return "", err
+	}
+
+	err = yaml.Unmarshal(metaData, meta)
+	if err != nil {
+		return "", err
+	}
+
+	chartFile := &chart.Metadata{
+		Name:        meta.Name,
+		Description: meta.Description,
+		// define Vela addon's type to be library in order to prevent installation of a common chart. Please refer to https://helm.sh/docs/topics/library_charts/
+		Type:       "library",
+		Version:    meta.Version,
+		AppVersion: meta.Version,
+		APIVersion: chart.APIVersionV2,
+		Icon:       meta.Icon,
+		Home:       meta.URL,
+		Keywords:   meta.Tags,
+	}
+
+	// save the Chart.yaml file in order to be compatible with helm chart
+	err = chartutil.SaveChartfile(filepath.Join(addonDictPath, chartutil.ChartfileName), chartFile)
+	if err != nil {
+		return "", err
+	}
+
+	ch, err := loader.LoadDir(addonDictPath)
+	if err != nil {
+		return "", err
+	}
+
+	dest, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+	archive, err := chartutil.Save(ch, dest)
```

Comment:
Is the package's name is {addonName}-{version}.tgz ?

Reply:
> Is the package's name is {addonName}-{version}.tgz ?

Yes. for example:
```
velaux-v1.3.3.tgz
fluxcd-1.1.0.tgz
```